clang/lib/Sema/SemaCodeComplete.cpp

```diff
@@ -4022,7 +4022,10 @@ CodeCompleteConsumer::OverloadCandidate::CreateSignatureString(
 
     std::string Name;
     llvm::raw_string_ostream OS(Name);
-    FDecl->getDeclName().print(OS, Policy);
+    auto const DeclName = (getKind() == CK_Lambda)
```

Comment:
For diagnostics, we print this exclusively based on type by printing "lambda  line:col" or something like that.  That said, `operator()` IS the name of this function, so I'd not want us to do away with that.

Reply:
This isn't the place for diagnostics. I think we're actually printing the DeclarationName for the function signature help, so we don't end up with an `operator()` in the lambda case.

<img width="314" alt="image" src="https://github.com/user-attachments/assets/006ac25d-b7cb-4895-b340-d9e5f9b05a37">

(This is what it looks like for lambdas currently)

I hope this could properly fall back to the previous logic if the lambda expression weren't declared with a name.